application/controllers/SurveysController.php

```diff
@@ -106,10 +106,13 @@ public function actionError()
      * @throws Twig_Error_Syntax
      * @throws WrongTemplateVersionException
      */
-    public function spitOutHtmlError(array $error, $oException = null)
+    public function spitOutHtmlError(array $error, $surveyId)
     {
-        // TODO: getGlobalSetting is DEPRECATED.
-        $oTemplate = Template::model()->getInstance(getGlobalSetting('defaulttheme'));
+        if ($surveyId) {
+            $oTemplate = Template::model()->getInstance('', $surveyId);
+        } else {
+            $oTemplate = Template::getLastInstance();
```

Comment:
If template is  not already set : we have the default template ?

We don't need `$oTemplate = Template::model()->getInstance(App()->getConfig('defaulttheme'));` ?

Reply:
Great !
Then : a plugin can set to another instance before !!
This is really great !